src/task/CreatePeerConnectionTask.ts

```diff
@@ -187,7 +194,23 @@ export default class CreatePeerConnectionTask extends BaseTask implements Remova
       width = cap.width as number;
       height = cap.height as number;
     }
-    tile.bindVideoStream(attendeeId, false, stream, width, height, streamId);
+    let externalUserId = this.context.realtimeController.realtimeExternalUserIdFromAttendeeId(
+      attendeeId
+    );
+    tile.bindVideoStream(attendeeId, false, stream, width, height, streamId, externalUserId);
+    if (!externalUserId) {
```

Comment:
@aburnson @XHatan I suggested we handle a case when the tile is added without `externalUserId`.

1. For customers using external user ID, it's a race condition. A video track is added ***before*** the `DefaultVolumeIndicatorAdapter` retrieves the associated external user ID from an audio stream.
2. For customers not using external user ID, SDK will still subscribe to presence changes. We don't know whether a customer is using external user ID for remote attendees.

Two questions:
- First of all, does this implementation make sense?
- For case (2), can we just check if an attendee is not present yet? i.e. If an attendee becomes present without external ID, this attendee will never have external ID in the future.
  ```
  if (!isAttendeePresentAlready) {
  ```

Reply:
Attendees will always have an external user ID since that is a required field. I'm OK with the workaround below (with one comment).